qiskit/circuit/library/standard_gates/u.py

```diff
@@ -92,9 +92,10 @@ def control(self, num_ctrl_qubits=1, label=None, ctrl_state=None):
     def _define(self):
         """Alias for U3 until U becomes a basis gate."""
         from qiskit.circuit.quantumcircuit import QuantumCircuit
+        from .u3 import U3Gate
         q = QuantumRegister(1, 'q')
         qc = QuantumCircuit(q)
-        qc.u3(self.params[0], self.params[1], self.params[2], q[0])
+        qc.append(U3Gate(self.params[0], self.params[1], self.params[2]), [q[0]])
```

Comment:
UGate should not be defined in terms of U3, since it is now a base gate. I think just having a `to_matrix` is sufficient and `_define` method can be removed.